sdks/fabric/interoperation-node-sdk/src/AssetManager.ts

```diff
@@ -666,11 +692,34 @@ const StartHTLCAssetLockListener = (
     assetId: string,
     recipientECert: string,
     lockerECert: string,
-    lockCallback: (c: Contract, d: string, t: string, i: string, r: string, l: string, v: string) => any,
+    lockCallback: (c: Contract, d: string, t: string, i: string, r: string, l: string, v: Hash, timeout: number) => any,
 ): void => {
     StartHTLCEventListener(contract, 'LockAsset', contractId, assetType, assetId, -1, recipientECert, lockerECert, lockCallback);
 }
 
+//TODO: Timed counterpart for timed AssetLockListener for reversion
+const assetLockCallbackTimed = (
+    contract: Contract, 
+    contractID: string, 
+    assetType: string, 
+    assetID: string, 
+    recipientECert: string, 
+    lockerECert: string, 
+    hashMechanism: Hash,
+    expiryTime: number
+): void => {
+    //Compare expiryTimeSec with currentTimeSec, which is number of milliseconds since epoch (L174)
+    const currTimeSecs = Math.floor(Date.now()/1000);  
+    if(expiryTime > currTimeSecs) {
+        claimAssetInHTLC(contract, assetType, assetID, recipientECert, hashMechanism);
```

Comment:
We should not have any claim logic here. Only unlocks should be automatic, upon the expiration of a timer. We need to support automatic claiming too, but using the event mechanism. Remember that the claimant is a different entity than the locker. This particular callback function is registered by the locker to fire whenever a lock event is detected. Adding a claim call here implies that the locker is itself trying to claim the asset it just locked.